yeat/cli/config.py

```diff
@@ -182,17 +183,22 @@ def to_string(self):
 class Assembler:
     def __init__(self, data, threads):
         self.label = data["label"]
-        if data["algorithm"] not in ALGORITHMS:
-            raise AssemblyConfigurationError(
-                f"Unsupported assembly algorithm '{data['algorithm']}'"
-            )
+        self.check_valid_algorithm(data)
         self.algorithm = data["algorithm"]
         self.samples = data["samples"]
         self.extra_args = data["extra_args"]
         self.threads = threads
         if self.algorithm == "canu":
             self.check_canu_required_params()
 
+    def check_valid_algorithm(self, data):
+        if data["algorithm"] not in ALGORITHMS:
+            message = f"Unsupported assembly algorithm '{data['algorithm']}'"
+            raise AssemblyConfigurationError(message)
+        if data["algorithm"] == "metamdbg" and platform not in ["linux", "linux2"]:
+            message = f"Assembly algorithm 'metaMDBG' can only run on Linux OS"
+            raise AssemblyConfigurationError(message)
```

Comment:
Platform check statement. If the user adds the algorithm `metamdbg` into the config but the OS platform is not `linux`, an `AssemblyConfigurationError` will popup.